### PR TITLE
[MINOR] Performance Test Bug Fixes

### PR DESCRIPTION
**Pull request by krishnakalyan3**
### Minor Performance test bugfixes
- [x] Add dense, sparse and all to args.
- [x] Use exact arguments from command line (not using folders that are not given in the command line)
- [x] Pylint Test
- [x] Test under different argument conditions

(singlenode)
https://gist.github.com/krishnakalyan3/84f37082d3e474d294860c071099f187

(spark_hybrid)
https://gist.github.com/krishnakalyan3/f559fb5980317340a53ecbe499166f73

```
./scripts/perftest/python/run_perftest.py --exec hybrid_spark --family binomial clustering multinomial regression1 regression2 stats1 stats2
./scripts/perftest/python/run_perftest.py --family binomial clustering multinomial regression1 regression2 stats1 stats2
./scripts/perftest/python/run_perftest.py --exec hybrid_spark --family clustering --mat-type sparse
./scripts/perftest/python/run_perftest.py --exec hybrid_spark --family clustering --mat-type all
./scripts/perftest/python/run_perftest.py --exec hybrid_spark --family stats1  --mat-type dense
./scripts/perftest/python/run_perftest.py --exec hybrid_spark --family stats1  --mat-type sparse
```

### CODE REVIEW

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1727/

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1730/

**Comment by krishnakalyan3** [comment]
ping @nakul02. Could you please review this PR

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1731/

File: scripts/perftest/python/utils.py [hunk 1]
```diff
@@ -301,4 +304,86 @@ def get_folder_metrics(folder_name, action_mode):
...
+    current_family: String
+    Current family being porcessed in this function
+
+    matrix_type: List
+    Type of matrix to generate dense, sparse, all
+
+    return: List
+    Return the list of right matrix types supported by the family
+    """
+    family_no_matrix_type = ['clustering', 'stats1', 'stats2']
```
Comment (nakul02) [inline scripts/perftest/python/utils.py]: Hardcoding which families do not need sparse datagen in a util function does not seem like a good idea to me.
Place it with the other config (in `run_perftest.py` where `ML_ALGO` and other configuration is specified).

File: scripts/perftest/python/utils.py [hunk 2]
```diff
@@ -301,4 +304,86 @@ def get_folder_metrics(folder_name, action_mode):
     except IndexError:
         intercept = 'none'
 
-    return mat_type, mat_shape, intercept
\ No newline at end of file
+    return mat_type, mat_shape, intercept
+
+
```
Comment (nakul02) [inline scripts/perftest/python/utils.py]: Maybe `matrix_type` should be `matrix_types` making it clear that it is plural.

**Comment by krishnakalyan3** [comment]
ping @nakul02 could you please review.

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1747/

**Comment by nakul02** [comment]
LGTM :+1: 
I shall merge.

**Pull request closed without merging**